src/com/company/Parse.java

```diff
@@ -0,0 +1,120 @@
+package com.company;
+
+import org.eclipse.cdt.core.dom.ast.ExpansionOverlapsBoundaryException;
+import org.eclipse.cdt.core.dom.ast.IASTNode;
+import org.eclipse.cdt.core.dom.ast.IASTTranslationUnit;
+import org.eclipse.cdt.core.dom.ast.gnu.cpp.GPPLanguage;
+import org.eclipse.cdt.core.index.IIndex;
+import org.eclipse.cdt.core.parser.*;
+import org.eclipse.cdt.internal.core.dom.parser.cpp.*;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+
+public class Parse {
+
+    private IASTTranslationUnit translationUnit; //current translation unit node
```

Comment:
This actually isn't used anywhere else, can you get rid of it? (and only keep `translationUnit` scoped in the constructor?)
Keeping variables scoped as narrowly as possible eases code reading and understanding, since it's clear that it can't be accessed (or worse, modified) elsewhere.